File: LeetCode/2626. Array Reduce Transformation/answer.ts
// 2626. Array Reduce Transformation
// https://leetcode.com/problems/array-reduce-transformation/description/

type Fn = (accum: number, curr: number) => number

function reduce(nums: number[], fn: Fn, init: number): number {
    let result = init;
    for (let i=0; i<nums.length; i++) {
        result = fn(result, nums[i])
    }
    return result
};

File: LeetCode/1470. Shuffle the Array/answer.py
# 1470. Shuffle the Array
# https://leetcode.com/problems/shuffle-the-array/

class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        result = [0] * (2 * n)
        for i in range(n):
            result[2*i] = nums[i]
            result[2*i+1] = nums[n+i]
        return result